### PR TITLE
systems.json gets automaticly downloaded

### LINKED ISSUE #15: systems.json is not automatically downloaded
In world explorer systems.json should be present at ./config/systems.json. 
This can be done manually but it should happen automaticly 

the endpoint of where it should get systems.json is https://api.spacetraders.io/v2/systems.json

### PR DESCRIPTION
**Pull request by Yvonne-Aizawa**
In world explorer systems.json gets automatically downloaded when it does not exist yet. 

this fixes #15

### CODE REVIEW

File: trader/src/parse_system.rs [hunk 1]
```diff
@@ -37,16 +39,49 @@ pub fn parse_json() -> Result<Vec<System>, std::io::Error> {
...
+            );
+            match download_result {
+                Ok(_) => {
+                    println!("File downloaded successfully");
+                    let mut file = File::open(file_path).unwrap();
+                    let mut contents = String::new();
+                    file.read_to_string(&mut contents)
+                        .expect("Failed to read file");
+                    let systems: Vec<System> =
+                        serde_json::from_str(&contents).expect("Failed to parse JSON");
```
Comment (Allstreamer) [inline trader/src/parse_system.rs]: These two expects should probably be replaced with a `.wrap_err("")?` operator

change the type the function is returning to `color_eyre::Result<Vec<System>>` and it should allow you to return any error types (color_eyre is like anyhow if you know that crate)

**Review by Allstreamer — CHANGES_REQUESTED** [review]

**Review by Yvonne-Aizawa — COMMENTED** [review]
seems correct to me (since i wrote it myself)

**Comment by Allstreamer** [comment]
One sec, gonna make some minor touch ups myself.

**Review by Allstreamer — APPROVED** [review]
Now errors don't panic the program but still log!

**Pull request merged**